File: Accounting/src/main/java/com/experiment/accounting/utils/ConstantReference.java
package com.experiment.accounting.utils;

import com.experiment.accounting.entity.base.BaseConstantEntity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ConstantReference {

    String name;
    String desc;
    Class<? extends BaseConstantEntity> clazz;
    List<ConstantReference> subConstants = new ArrayList<>();
}

File: Inventory/src/main/kotlin/com/experiment/inventory/utils/ConstantReference.kt
package com.experiment.inventory.utils

import com.experiment.inventory.entity.base.BaseConstantEntity
import kotlin.reflect.KClass
import kotlin.reflect.jvm.internal.impl.resolve.calls.inference.CapturedType

class ConstantReference<T:BaseConstantEntity> {
    var name: String? = null
    var desc: String? = null
    var clazz: KClass<T>? = null
    var subConstants: List<ConstantReference<T>> = ArrayList()
}